### PR TITLE
Documentation of JSON API, Best Effort Power Cap. Edit Example links

### LINKED ISSUE #154: Documentation: JSON API, Best Effort Power Capping, and Updated Build Info
Just a placeholder issue for the documentation update coming from my end. 
We need to add a page on how to use the JSON API to interact with higher level tools, and also our notes on what Best Effort Power Capping implies in the current version. Also need to add the documentation on how to compile for with and without NVML monitoring, as that can impact some outputs. 

- [x] Remove edits to Examples.rst from this PR (changes integrated with `fff9829`)
- [x] Add JSON API example

### PR DESCRIPTION
**Pull request by tpatki**
WIP.

Fixes #154

### CODE REVIEW

File: src/docs/sphinx/VariorumTools.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
+.. # Copyright 2019-2020 Lawrence Livermore National Security, LLC and other
```
Comment (slabasan) [inline src/docs/sphinx/VariorumTools.rst:1]: License year should just be 2021 (since added this year)

File: src/docs/sphinx/VariorumAPI.rst [hunk 2]
```diff
@@ -12,3 +12,66 @@ The top-level API for Variorum consists of the following calls:
...
+power cap, a best-effort power cap is determined in sofware to provide an 
+easier interface for higher-level tools (e.g. Flux, Kokkos, etc). 
+
+For example, while IBM Witherspoon inherently provides the ability to set a 
+node-level power cap in watts in hardware through its OPAL infrastructure, 
+Intel architectures currently do not support a direct node level power cap through
+MSRs. Instead, on Intel architectures, fine-grained CPU and DRAM level power 
+caps can be dialed in using MSRs. Note that IBM Witherspoon does not provide
+fine-grained capping for CPU and DRAM level, but allows for a power-shifting
+ratio between the CPU and GPU components on a socket (see IBM documentation.)
```
Comment (slabasan) [inline src/docs/sphinx/VariorumAPI.rst:69]: Add link to IBM documentation:
(see `IBM documentation <https://variorum.readthedocs.io/en/latest/IBM.html>`_).

File: src/docs/sphinx/VariorumTools.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
+.. # Copyright 2019-2021 Lawrence Livermore National Security, LLC and other
```
Comment (slabasan) [inline src/docs/sphinx/VariorumTools.rst:1]: License should start with 2021

File: src/docs/sphinx/VariorumTools.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+.. #
+.. # SPDX-License-Identifier: MIT
+
+
+Using Tools with Variorum
+=========================
+
+The JSON API enables Variorum to interface with higher-level system software.
+Current integration efforts include a Kokkos connector for power monitoring and
+a Flux power managment module.
```
Comment (slabasan) [inline src/docs/sphinx/VariorumTools.rst:12]: managment --> management

File: src/docs/sphinx/VariorumAPI.rst [hunk 5]
```diff
@@ -12,3 +12,66 @@ The top-level API for Variorum consists of the following calls:
...
+cross-architectural build, similar to Variorum's ``variorum_get_node_power()`` API.
+ 
+Best Effort Power Capping
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We support setting best effort node power limits in a vendor-neutral manner. 
+This interface has been developed from the point of view of higher-level tools 
+that utilize Variorum on diverse architectures and need to make node-level 
+decisions. When the underlying hardware does not directly support a node-level 
+power cap, a best-effort power cap is determined in sofware to provide an
```
Comment (slabasan) [inline src/docs/sphinx/VariorumAPI.rst:60]: sofware --> software

File: src/docs/sphinx/VariorumTools.rst [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+Links to each of these frameworks can be found 
+below. Note that these integrations with Variorum are in early development 
+stages and are expected to be updated to support more features and tests.
+Upcoming integration also includes developing a Variorum service for Caliper. 
+  
+    * Kokkos connector: https://github.com/kokkos/kokkos-tools/tree/develop/profiling/variorum-connector
+    * Flux Power Manager Module: https://github.com/rountree/flux-power-mgr 
+
+In order for tools to interact with Variorum, a simple Jansson based parser
```
Comment (slabasan) [inline src/docs/sphinx/VariorumTools.rst:21]: Should Jansson be all caps here?

File: src/docs/sphinx/VariorumTools.rst (hunk 3 above)
Comment (tpatki) [reply src/docs/sphinx/VariorumTools.rst:1]: Bit confused. I was following what you did in your license update...Should this be "2021" only or "2021-2019"? That'd seem odd.

Comment (slabasan) [reply src/docs/sphinx/VariorumTools.rst]: The start year should be the year the file was added to the repo. So this file next year will be updated to `2021-2022`, and other files will be `2019-2022`, etc.

Comment (tpatki) [reply src/docs/sphinx/VariorumTools.rst]: Got it, so I made the right change then. Take a look at the new commits.

Comment (slabasan) [reply src/docs/sphinx/VariorumTools.rst]: Yep the header looks correct!

**Pull request merged**